tee_supplicant/src/tee_supplicant.c

```diff
@@ -52,6 +72,100 @@ static int receive_request(const struct device *dev, struct tee_supp_msg *ts_req
 	return rc;
 }
 
+static struct suppl_shm *find_shm(uint64_t id)
+{
+	struct suppl_shm *curr = NULL;
+	uint64_t mem_id;
+
+	if (id) {
+		mem_id = (uint64_t)((struct tee_shm *)id)->addr;
+
+		tee_supp_mutex_lock(&shm_mutex);
+
+		curr = shm_head;
+		while (curr && curr->id != mem_id) {
+			curr = curr->next;
+		}
+
+		tee_supp_mutex_unlock(&shm_mutex);
+	}
+
+	return curr;
+}
+
+static struct suppl_shm *pop_shm(uint64_t id)
+{
+	struct suppl_shm *curr = NULL;
+	struct suppl_shm *prev = NULL;
+	uint64_t mem_id;
+
+	if (id) {
+		mem_id = (uint64_t)((struct tee_shm *)id)->addr;
+
+		tee_supp_mutex_lock(&shm_mutex);
+
+		curr = shm_head;
+		if (!curr) {
+			goto out;
+		}
+
+		if (curr->id == mem_id) {
+			shm_head = curr->next;
+			goto out;
+		}
+
+		do {
+			prev = curr;
+			curr = curr->next;
+			if (!curr)
+				goto out;
+		} while (curr->id != mem_id);
+		prev->next = curr->next;
+
+out:
+		tee_supp_mutex_unlock(&shm_mutex);
+	}
+
+	return curr;
+}
+
+static void push_shm(struct suppl_shm *shm)
```

Comment:
"push" implies stack-like operation. I believe this is not the case. How about "remember_shm"?

Reply:
Renamed